product_template_pricelist_button/models/product_template.py

```diff
@@ -0,0 +1,31 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import models
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    def open_pricelist_rules(self):
+        action = super().open_pricelist_rules()
+        self.ensure_one()
+        domain = ['|','|',
+            ('product_tmpl_id', '=', self.id),
+            ('product_id', 'in', self.product_variant_ids.ids),
+            ('categ_id','parent_of',self.categ_id.id),
+            ]
```

Comment:
This could be improved in order to prevent pasting existing code and overwriting it.

[Original code](https://github.com/odoo/odoo/blob/178ed5f2300241e2db9ced8ab89ba52852000ffa/addons/product/models/product_template.py#L613-L615) sets a `A or B` condition for the domain (where A refers to `product_tmpl_id` condition and B for the `product_id` one).
We need adding a new C condition, so the final condition should be `A or B or C`, that is the same as `(A or B) or C`... adding "`or C`" to the existing condition (where C is the `categ_id` condition) indeed. This could be achieved using `expression.OR` method (see e.g. https://github.com/OCA/product-attribute/blob/e6a73b3575058e395c24a16886bba90d5189ed22/product_pricelist_direct_print/wizards/product_pricelist_print.py#L311-L313).

@IriaAlonso if you don't understand how to use it, please call @ChristianSantamaria